### ISSUE TITLE
hs.base64.decode seems to truncate by one character

### ISSUE THREAD

**Issue by cmsj** [Hammerspoon/hammerspoon#469]
```
> hs.base64.encode("lolwat")
bG9sd2F

> hs.base64.decode("bG9sd2F")
lolwa
```

**Comment by asmagill** [comment]
Try it with _encode and _decode... without having had a chance to look at the code yet, Im going to go out on a limb and suggest that the problem is in the code which tries to "cleanup" base64 encoded data which has linefeeds for better presentation, transport, etc.

**Comment by asmagill** [comment]
Yeah, replace module.encode in the base64 init.lua with this... should fix it:

module.encode = function(data, width)
    local _data = module._encode(data)
    if width then
        local _hold, i, j = _data, 1, width
        _data = ""
        repeat
            _data = _data.._hold:sub(i,j).."\n"
            i = i + width
            j = j + width
        until i > #_hold
        return _data:sub(1,#_data - 1)
    else
        return _data
    end
end


If you don't get a chance, I'll do so later today.

**Comment by cmsj** [comment]
Closed by d1047c9
